### PR TITLE
fix(connector): validate connector name before converting ssl certs

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes https://emqx.atlassian.net/browse/EMQX-11336

See also: https://github.com/emqx/emqx/pull/11540

### <samp>🤖 Generated by Copilot at d108dd2</samp>

This pull request adds a new feature to validate the bridge and connector names using a regular expression and fixes a bug in the V1 HTTP API for creating bridges with SSL options. It updates the schema, the `emqx_bridge` and `emqx_connector` modules, and the corresponding test suites. It also adds new test cases for different bridge and connector types and configuration paths.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_connector/src/emqx_connector.erl [hunk 1]
```diff
@@ -458,3 +460,55 @@ ensure_no_channels(Configs) ->
         {error, Reason, _State} ->
             {error, Reason}
     end.
+
+to_bin(A) when is_atom(A) -> atom_to_binary(A, utf8);
+to_bin(B) when is_binary(B) -> B.
+
+validate_connector_name(ConnectorName0) ->
+    ConnectorName = to_bin(ConnectorName0),
+    case re:run(ConnectorName, ?MAP_KEY_RE, [{capture, none}]) of
```
Comment (zmstone) [inline apps/emqx_connector/src/emqx_connector.erl:469]: there is `emqx_resource:validate_name/1`

Comment (thalesmg) [reply apps/emqx_connector/src/emqx_connector.erl]: :heavy_check_mark:

File: apps/emqx_bridge_azure_event_hub/test/emqx_bridge_azure_event_hub_producer_SUITE.erl [hunk 2]
```diff
@@ -317,3 +317,28 @@ t_same_name_azure_kafka_bridges(AehConfig) ->
         end
     ),
     ok.
+
+t_create_with_bad_name(Config0) ->
```
Comment (sstrigler) [inline apps/emqx_bridge_azure_event_hub/test/emqx_bridge_azure_event_hub_producer_SUITE.erl:321]: Not so sure about this, why test it here too? It's been reported for AEH, but the problem has nothing to do with it. Future people might start to wonder why this is tested here. Do we want to test that with every bridge?

Certainly we want to test whether the compat layer (i.e. using a v2 bridge works through the old `/bridges` API endpoint), but isn't that done elsewhere too?

Anyway, maybe leave a comment at least, and explain why this is here.

File: apps/emqx_connector/src/emqx_connector.erl [hunk 3]
```diff
@@ -458,3 +468,51 @@ ensure_no_channels(Configs) ->
...
+    catch
+        throw:Error ->
+            {error, Error}
+    end.
+
+validate_connector_name_in_config(Path) ->
+    case Path of
+        [?ROOT_KEY, _ConnectorType, ConnectorName] ->
+            validate_connector_name(ConnectorName);
+        _ ->
```
Comment (sstrigler) [inline apps/emqx_connector/src/emqx_connector.erl:488]: Is this something to expect? When? Otherwise, wouldn't it be better to just crash?

Comment (thalesmg) [reply apps/emqx_connector/src/emqx_connector.erl:488]: True, I don't think this should be reached.  I'll removed it.

Comment (thalesmg) [reply apps/emqx_connector/src/emqx_connector.erl:488]: Actually, no, it may happen if someone does `emqx_conf:update([connectors, type], ...)`.

**Review by kjellwinblad — APPROVED** [review]

**Review by ieQu1 — APPROVED** [review]

**Pull request merged**